web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue

```diff
@@ -1,22 +1,944 @@
+                <balance-token-card
+                    v-for="item in mostRecentTransaction"
+                    :key="item.id"
+                    :billing-item="item"
+                    :show-add-funds="showAddFunds"
+                    @showTransactions="toggleTransactionsTable"
+                    @showAddFunds="toggleShowAddFunds"
+                />
+            </div>
+            <div v-else>
+                <add-token-card
+                    :total-count="transactionCount"
+                    @showAddFunds="toggleShowAddFunds"
+                />
+            </div>
+            <div v-for="card in creditCards" :key="card.id" class="payments-area__container__cards">
+                <CreditCardContainer
+                    :credit-card="card"
+                    @remove="removePaymentMethodHandler"
+                />
+            </div>
+            <div class="payments-area__container__new-payments">
+                <div v-if="!isAddingPayment" class="payments-area__container__new-payments__text-area">
+                    <span class="payments-area__container__new-payments__text-area__plus-icon">+&nbsp;</span>
+                    <span 
+                        class="payments-area__container__new-payments__text-area__text"
+                        @click="addPaymentMethodHandler"
+                    >Add New Payment Method</span>
+                </div>
+                <div v-if="isAddingPayment">
+                    <div class="close-add-payment" @click="closeAddPayment">
+                        <CloseCrossIcon />
+                    </div>
+                    <div class="payments-area__create-header">Credit Card</div>
+                    <div class="payments-area__create-subheader">Add Card Info</div>
+                    <StripeCardInput
+                        ref="stripeCardInput"
+                        class="add-card-area__stripe stripe_input"
+                        :on-stripe-response-callback="addCard"
+                    />
+                    <div
+                        v-if="!isAddCardClicked"
+                        class="add-card-button"
+                        @click="onConfirmAddStripe"
+                    >
+                        <img
+                            v-if="isLoading"
+                            class="payment-loading-image"
+                            src="@/../static/images/account/billing/loading.gif"
+                            alt="loading gif"
+                        >
+                        <SuccessImage
+                            v-if="isLoaded"
+                            class="payment-loaded-image"
+                        />
+                        <span class="add_card_button_text">Add Credit Card</span>
+                    </div>
+                </div>
+            </div>
+        </div>
+        
+        <div v-if="isRemovePaymentMethodsModalOpen || isChangeDefaultPaymentModalOpen" class="edit_payment_method">
```

Comment:
nit - is this v-if necessary? My understanding is this code is structured like
```
<div v-if="a || b">
    <div v-if="a">...</div>
    <div v-if="b">...</div>
</div>
```
but wouldn't this be sufficient?
```
<div v-if="a">...</div>
<div v-if="b">...</div>
```
mainly I'm just curious. This doesn't block my approval.

Reply:
it is necessary because the parent v-if contains 2 different child component inside it that each have their own v-if

Reply:
oh, I didn't notice in my initial review that the parent component was necessary for styling